keystorage/interlock/src/main/java/tech/pegasys/signers/interlock/InterlockSessionFactoryProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signers.interlock;
+
+import tech.pegasys.signers.interlock.vertx.InterlockSessionFactoryImpl;
+
+import java.nio.file.Path;
+
+import io.vertx.core.Vertx;
+import org.apache.commons.lang3.StringUtils;
+
+public class InterlockSessionFactoryProvider {
+  private static final String HTTP_CLIENT_TIMEOUT_ENV = "INTERLOCK_CLIENT_TIMEOUT";
+  private static final int DEFAULT_TIMEOUT_MS = 5000;
+  private static final int HTTP_CLIENT_TIMEOUT_MS = timeoutMS();
+
+  public static InterlockSessionFactoryImpl newInstance(
+      final Vertx vertx, final Path serverWhitelist) {
+    return new InterlockSessionFactoryImpl(vertx, serverWhitelist, HTTP_CLIENT_TIMEOUT_MS);
+  }
+
+  private static int timeoutMS() {
```

Comment:
should the MS both be capitalised - maybe Ms?

Reply:
I've modified it to return `Duration`, hence method name is changed to `timeoutDuration`